internal/vxproto/ws_server.go

```diff
@@ -299,48 +304,64 @@ type getConnectionPolicyMiddlewareResult struct {
 	AgentConnectionInfo *AgentConnectionInfo
 }
 
-func getConnectionPolicy(ctx context.Context, urlPath string, agentID string, idFetcher AgentIDFetcher, policyManager ConnectionPolicyManager) (*getConnectionPolicyMiddlewareResult, error) {
-	agentType, err := getAgentTypeFromURL(urlPath)
+func getConnectionPolicy(
+	ctx context.Context,
+	urlPath string,
+	id string,
+	idFetcher AgentIDFetcher,
+	policyManager ConnectionPolicyManager,
+) (*getConnectionPolicyMiddlewareResult, error) {
+	connType, err := getConnectionTypeFromURL(urlPath)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get the agent type from the URL path: %w", err)
+		return nil, fmt.Errorf("failed to get the connection type from the URL path: %w", err)
 	}
-	if !policyManager.IsConnectionAllowed(agentType) {
-		return nil, fmt.Errorf("connection for the given agent type (%d) is forbidden: %w", agentType, ErrEndpointBlocked)
+	if !policyManager.IsConnectionAllowed(connType) {
+		return nil, fmt.Errorf("connection for the given type (%d) is forbidden: %w", connType, ErrEndpointBlocked)
 	}
-	connInfo, err := fetchAgentConnectionInfo(ctx, idFetcher, agentID)
+	fetchType := "agent"
+	if connType == Aggregate {
+		fetchType = "group"
```

Comment:
Does it mean that it won't be possible to reuse same logic for "policy" aggregates? Maybe then name "aggregate" is misleading and we should just use "group"?

Reply:
I'm not sure that using the same logic for policy is a good idea because policy may linked to multiple groups so it will be multiple lua states.
It could solved though multiple "aggregate" connections from JS part with next steps:
1. user opens policy module view;
2. the view form enumerates all groups and connects to them via vxproto;
3. protoAPI object contains multiple source tokens and developer decides how connections will be using in browser module.
Of course, protoAPI object can use unified packet types callbacks for all connections and after use callback split it by source token on the module side.